src/core/Akka/Actor/Futures.cs

```diff
@@ -164,7 +164,7 @@ public static Task<T> Ask<T>(this ICanTell self, Func<IActorRef, object> message
             //The future actor needs to be registered in the temp container
             provider.RegisterTempActor(future, path);
             var message = messageFactory(future);
-            self.Tell(message, future);
+            future.DeliverAsk(message, self);
```

Comment:
Made the `.Tell` implementation internal to the `FutureActorRef<T>` so it's possible to generate telemetry here (and do things like filter which `Ask<T>` operations should be traced and which ones should not.)